src/devices/fineoffset.c

```diff
@@ -317,6 +317,76 @@ static int fineoffset_WH24_callback(r_device *decoder, bitbuffer_t *bitbuffer)
     return 1;
 }
 
+/**
+Fine Offset Electronics WH0290 Wireless Air Quality Monitor
+Also: Ambient Weather PM25
+
+The sensor sends a package each ~10m. The bits are PCM modulated with Frequency Shift Keying.
+
+Data layout:
+    aa 2d d4 42 cc 41 9a 41 ae c1 99 9
+             FF DD ?P PP ?A AA CC BB
+
+- F: 8 bit Family Code?
+- D: 8 bit device id?
+- ?: 2 bits ?
+- P: 14 bit PM2.5 reading in ug/m3
+- ?: 2 bits ?
+- A: 14 bit PM10 reading in ug/m3
+- ?: 8 bits ?
+- C: 8 bit CRC checksum of the previous 6 bytes
+- B: 8 bit Bitsum (sum without carry, XOR) of the previous 7 bytes
+
+*/
+static int fineoffset_WH0290_callback(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    data_t *data;
+    uint8_t const preamble[] = {0xAA, 0x2D, 0xD4};
+    uint8_t b[8];
+    unsigned bit_offset;
+
+    bit_offset = bitbuffer_search(bitbuffer, 0, 0, preamble, sizeof(preamble) * 8) + sizeof(preamble) * 8;
+    if (bit_offset + sizeof(b) * 8 > bitbuffer->bits_per_row[0]) {  // Did not find a big enough package
+        if (decoder->verbose)
+            bitbuffer_printf(bitbuffer, "Fineoffset_WH0290: short package. Row length: %u. Header index: %u\n", bitbuffer->bits_per_row[0], bit_offset);
+        return DECODE_ABORT_LENGTH;
+    }
+    bitbuffer_extract_bytes(bitbuffer, 0, bit_offset, b, sizeof(b) * 8);
+
+    // Verify checksum, same as other FO Stations: Reverse 1Wire CRC (poly 0x131)
+    uint8_t crc = crc8(b, 6, 0x31, 0x00);
+    uint8_t checksum = 0;
+    for (unsigned n = 0; n < 7; ++n) {
+        checksum += b[n];
+    }
+    if (crc != b[6] || checksum != b[7]) {
+        if (decoder->verbose) {
+            fprintf(stderr, "Fineoffset_WH0280: Checksum error: %02x %02x\n", crc, checksum);
+        }
+        return DECODE_FAIL_MIC;
+    }
+
+    // Decode data
+    uint8_t family    = b[0];
+    uint8_t id        = b[1];
+    int pm25          = (b[2] & 0x3f) << 8 | b[3];
+    int pm100         = (b[4] & 0x3f) << 8 | b[5];
+
+
+    /* clang-format off */
+    data = data_make(
+            "model",            "",             DATA_STRING, _X("Fineoffset-WH0290","Fine Offset Electronics, WH0290"),
+            "id",               "ID",           DATA_INT,    id,
+            "25_particles",     "2.5 micron particles",  DATA_FORMAT, "%i ug/m3", DATA_INT, pm25/10,
+            "100_particles",     "10 micron particles",  DATA_FORMAT, "%i ug/m3", DATA_INT, pm100/10,
```

Comment:
Conventional names are "coarse particulate matter", (PM10) and "fine particulate matter" (PM2.5), maybe we can find a better key based on that. (e.g. should be "pm10_ug_m3" -- the dot in 2.5 could be a problem though…)

Reply:
Maybe pm2_5_ug_m3 and pm10_ug_m3 to avoid the dot?

Reply:
Here's what those look like:
```
time      : @0.084044s
model     : Fine Offset Electronics, WH0290        ID        : 204
2.5um Fine Particulate Matter: 9 ug/m3             10um Coarse Particulate Matter: 10 ug/m3            Integrity : CHECKSUM
```
```
{"time" : "@0.084044s", "model" : "Fine Offset Electronics, WH0290", "id" : 204, "pm2_5_ug_m3" : 9, "pm10_ug_m3" : 10, "mic" : "CHECKSUM"}
```

Reply:
I Pushed the new names and descriptions

Reply:
fwiw, in the weewx air-quality schema we have been using pm_10_0, pm_2_4, and pm_1_0

Reply:
Seems a good scheme. Please add the `_0` onto the `10` for uniform looks.

Reply:
Oh, too late. I'll commit that change directly :)